src/main/java/de/tum/cit/aet/artemis/programming/repository/BuildJobRepository.java

```diff
@@ -104,4 +104,20 @@ SELECT COUNT(b)
             WHERE e.id IN :exerciseIds
             """)
     long countBuildJobsByExerciseIds(@Param("exerciseIds") List<Long> exerciseIds);
+
+    @Query("""
+            SELECT b
+            FROM BuildJob b
+            WHERE b.exerciseId = :exerciseId AND b.buildStatus = 'SUCCESSFUL'
+            ORDER BY b.buildStartDate DESC
+            LIMIT :limit
+            """)
+    List<BuildJob> fetchSuccessfulBuildJobsByExerciseIdWithLimit(@Param("exerciseId") Long exerciseId, @Param("limit") int limit);
+
+    @Query("""
+            SELECT COUNT(b)
+            FROM BuildJob b
+            WHERE b.exerciseId = :exerciseId AND b.buildStatus = 'SUCCESSFUL'
+            """)
+    long fetchSuccessfulBuildJobCountByExerciseId(@Param("exerciseId") Long exerciseId);
```

Comment:
Do we always need to go over all successful builds? 
Can we be more restrictive here? For large exercises in EIST (2000+ students) this can really grow to a quite large number that then needs to be handled in `updateExerciseBuildDuration` - from my experience most students will have 4-10 pushes and some students regularly pass the 30 commits mark 😅

Reply:
This will only be used to get a count of the total successful build jobs. We need to this to check how many new build jobs were processed since we last updated the average duration. For updating the duration, we use the query above with a limit